### ISSUE TITLE
WebSocket client error: Failed reading HTTP status line

### ISSUE THREAD

**Issue by steakpowered** [dechamps/LGTVDeviceListener#2]
After updating to version 03.30.16, this error comes up and no longer switches input.  Verbose doesn't display any additional information.

[info   ] Switching LGTV to input: HDMI_1
[ERROR  ] In device event listener: WebSocket client error: Failed reading HTTP status line from ws://xxx.xxx.xxx.xxx:3000 (retries: 0, wait time: 0.000000 ms, HTTP status: 0)

**Comment by dechamps** [comment]
In the latest firmware update LG appears to have changed the way clients can connect to the TV, as evidenced by various other tools being forced to make changes (for example [bscpylgtv](https://github.com/chros73/bscpylgtv/commit/5ee8d91876d0f6d77835008637c98daa6203af0a), [LGTVCompanion](https://github.com/JPersson77/LGTVCompanion/commit/c320fc0922c6dea7df06b682f067661540b52466#diff-4e3239d82382f5afc7aaf2936a8abdce0a81396b3ccd56bc1507ae52f7d68cafR492)).

Looking at the changes, it looks like the WebSocket port has changed from 3000 to 3001, and the use of SSL/TLS is now required.

Interestingly, it could very well be that LGTVDeviceListener can already do this with no changes, as I suspect it's already built with SSL support, if memory serves me right.

Can you try the following command and report back:

```
LGTVDeviceListener.exe --verbose --url wss://xxx.xxx.xxx.xxx:3001
```

It's basically the same as before except the use of `wss:` instead of `ws:` and port 3000 is changed to 3001.

I'm not sure it will work because it might try to verify certificates which will likely fail. In that case I'd need to take a closer look and make some changes.

**Comment by dechamps** [comment]
I just tried the above myself and yeah, as expected it doesn't quite work:

```
[ERROR  ] FATAL: WebSocket client error: Unable to connect to pegasus-lgtv.home.edechamps.fr on port 3001, error: error in handshake : X509 - Certificate verification failed, e.g. CRL, CA or signature check failed (retries: 0, wait time: 0.000000 ms, HTTP status: 0)
```

I'll need to find a way to disable certificate verification.

**Comment by dechamps** [comment]
This is fixed in [LGTVDeviceListener 0.2](https://github.com/dechamps/LGTVDeviceListener/releases/tag/LGTVDeviceListener-0.2).

Note the [updated instructions](https://github.com/dechamps/LGTVDeviceListener/blob/LGTVDeviceListener-0.2/README.md) which use a different URL - `wss://xxx.xxx.xxx.xxx:3001` should be used instead of `ws://xxx.xxx.xxx.xxx:3000` for LGTVDeviceListener to work with latest LG firmware versions. If you are running as a service, you will want to [delete the service and then re-create it](https://github.com/dechamps/LGTVDeviceListener/blob/master/README.md#running-as-a-windows-service) with the new URL.

**Comment by steakpowered** [comment]
Thanks for fixing this so quickly!
